### ISSUE TITLE
Darc does not update packages which were added in an existing PR

### ISSUE THREAD

**Issue by ericstj** [dotnet/arcade-services#3562]
See https://github.com/dotnet/efcore/pull/33810/commits/79197f7b372fe0d4e0e8f9072f7ee52097e705cc

I had added two packages to Version.Details.xml.  Darc then modified the PR with a new version of runtime but it didn't update the packages I added. 

Probably Darc is looking at the target branch's file rather than the PR branch's file.

**Comment by mmitche** [comment]
> Probably Darc is looking at the target branch's file rather than the PR branch's file.

Yes, Maestro is reading from the target branch initially when creating the branch, and either it keeps looking at the target branch, or it has a list of previously updated packages and only uses those for future updates. Can't remember which.

We can probably make a partial fix by reading dependencies from the PR branch if an existing PR is already open. A 'full' fix would be more difficult. A common scenario that happens is that a PR is opened for updates, someone adds a new dependency in a different PR, and both PRs merge. Often they don't conflict so you can get incoherency that way. Maestro would need to be aware of the PR target branch's current state and ensure that the dependency set wasn't changed. If it was, then it needs to restart the update process, pulling in a new build.

**Comment by premun** [comment]
We made a related small change that reads the versions from the PR head branch and I think it's possible to make it behave this way. The fix itself can be quite simple but I am worried about not covering scenarios such as people removing a dependency in the PR or similar which makes this a bit more complicated of a change. But we will see if we can do this one.

**Comment by mmitche** [comment]
There are for sure a number of corner cases here in order to get things perfect. Reading from the PR branch is an incremental improvement and still worth it IMO.

**Comment by premun** [comment]
To be done here: Verify if this is still the behaviour or it has been fixed.
